lib/web/apiserver_test.go

```diff
@@ -2393,6 +2401,91 @@ func TestWebSessionsRenewAllowsOldBearerTokenToLinger(t *testing.T) {
 	require.True(t, trace.IsAccessDenied(err))
 }
 
+type testCloudModules struct {
+	modules.Modules
+}
+
+func (m *testCloudModules) Features() modules.Features {
+	return modules.Features{
+		Cloud: true, // Explicily turn on cloud feature.
+	}
+}
+
+// TestReceivingAccountRecoveryCodesWhenChangingAuthns tests for following:
+//  - Recovery codes are not returned for usernames that are not emails
+//  - Recovery codes are returned for usernames that are valid emails
+func TestReceivingAccountRecoveryCodesWhenChangingAuthenticationsWithToken(t *testing.T) {
```

Comment:
Suggestion: Begin the test name with the method being tested, add scenarios and outcomes to the end.

```suggestion
func TestChangeUserAuthentication_recoveryCodesReturnedForCloud(t *testing.T) {
```